ste/xfer-remoteToLocal.go

```diff
@@ -73,6 +73,17 @@ func remoteToLocal(jptm IJobPartTransferMgr, p pipeline.Pipeline, pacer pacer, d
 		}
 	}
 
+	if jptm.MD5ValidationOption() == common.EHashValidationOption.FailIfDifferentOrMissing() {
+		// We can make a check early on MD5 existence and fail the transfer if it's not present.
+		// This will save hours in the event a user has say, a several hundred gigabyte file.
+		if len(info.SrcHTTPHeaders.ContentMD5) == 0 {
+			jptm.LogDownloadError(info.Source, info.Destination, errExpectedMd5Missing.Error(), 0)
+			jptm.SetStatus(common.ETransferStatus.Failed())
+			jptm.ReportTransferDone()
+			return
+		}
+	}
```

Comment:
Should the previous error location be removed? As it should be an unreachable condition since it's caught here first. 

Or perhaps changed to an `unexpected condition` style error?

Reply:
I'll promote the old check to a panic, since it's now in a place that shouldn't even be reached to start with.